engines/wintermute/base/sound/base_sound_manager.cpp

```diff
@@ -97,6 +97,12 @@ BaseSoundBuffer *BaseSoundMgr::addSound(const Common::String &filename, Audio::M
 		return nullptr;
 	}
 
+	if (filename.empty()) {
+		// At least one game, Bickadoodle, calls playSound with an empty filename, see #6594
+		BaseEngine::LOG(0, "addSound called with empty filename");
```

Comment:
@somaen: If I add a breakpoint in CBSoundBuffer::LoadFromFile I can reproduce in the original engine, so it's either a script bug or... a bugfeature of the original script interpreter.

In the original, of course, it just returns an E_FAIL when it can't find the file, instead of segfaulting.

I removed the `return nullptr` bit because it seems to me the segfaulting problem is best fixed in #733, I was originally wary of being that heavy handed, but after spending enough time with it and testing for a few edge cases I think it's solid enough.

I guess logging the anomaly would not hurt in case some similar bug pops up in the future, though?